ui: Reset 'lazyredraw' during ui_refresh.

justinmk opened the pull request:
There might be a more elegant way to convince `redrawing()` during this sequence.

References #6202
References https://github.com/neovim/neovim/pull/6202#issuecomment-284379503
References #3929 #5692 #4884 #6157

@choco can you see if this fixes the resize issue for you?

justinmk committed:
ui: disable 'lazyredraw' during ui_refresh.

Could also try `do_redraw = true` instead of save/restore `p_lz`, but
the nice thing about save/restore of `p_lz` is that it is "atomic".
The semantics of `do_redraw` are not clear to me.

Closes #4884
References #6202
References https://github.com/neovim/neovim/pull/6202#issuecomment-284379503
References #3929 #5692 #6157
References #5866

choco commented:
@justinmk Yeah, I was already using this patch and that fixes it for me!

justinmk commented:
Oh. Did you post it somewhere?

choco commented:
Oh, I though I posted it as an edit in the previous thread but I must have forgot sorry :/, but here it was https://gist.github.com/choco/c9a4f57cfcc93ac4d300db95a19747f3 I just copied the code from the ex command redraw function.
I'm just curious why this would be necessary, but I don't know enough about the ui code.

justinmk commented:
> I'm just curious why this would be necessary, but I don't know enough about the ui code.

`ex_redraw` does something similar, but I guess your question is more like "Why doesn't the system already know that 'lazyredraw' is irrelevant here?" I agree, that's what I was getting at in regards to `redrawing()`.

justinmk committed:
doc: screen_resize(): remove mention of "mustset"

"mustset" is from Vim term.c:set_shellsize(), not relevant here: we
behave as if mustset=true always.
